include/h2o/busypoll.h

```diff
@@ -0,0 +1,50 @@
+#ifndef h2o__busypoll_h
+#define h2o__busypoll_h
+
+#if !H2O_USE_EPOLL_BUSYPOLL
+#error "this file may be included only on linux and when ynl support is available"
+#endif
+
+#include <stdint.h>
+
+#define BP_MODE_OFF      0
+#define BP_MODE_SUSPEND  1
+#define BP_MODE_BUSYPOLL 2
```

Comment:
seems like this might be a good candidate for an enum instead of defines? but i suppose that is a stylistic choice left to the maintainer

Reply:
thank you, it used to be an enum, but i needed to make the epoll loop implementation aware of the mode to set `max_wait` appropriately and i did not want to include busypoll.h, we can still do that though.

Reply:
i dont really have a strong preference one way or the other -- just a thing i noticed and commented on. up to you if you want to make it an enum or not

Reply:
Could you elaborate the hesitation behind including busypoll.h from epoll.c.h?

It could very well be the case that something has changed since, but I think we can include files with checking the feature flags. Even epoll.c.h itself is included under the condition of `H2O_USE_EPOLL` being set.

Reply:
changed back to enum in 13a2745db394da3242e72d767aa697a7a731ce01